services/src/test/scala/cromwell/services/auth/impl/GithubAuthVendingActorSpec.scala

```diff
@@ -0,0 +1,120 @@
+package cromwell.services.auth.impl
+
+import akka.actor.{ActorRef, ActorSystem, Props}
+import akka.testkit.TestProbe
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.core.TestKitSuite
+import cromwell.services.auth.GithubAuthVending.{
+  GithubAuthRequest,
+  GithubAuthTokenResponse,
+  GithubAuthVendingFailure,
+  NoGithubAuthResponse
+}
+import cromwell.services.auth.ecm.EcmService
+import cromwell.services.auth.impl.GithubAuthVendingActorSpec.TestGithubAuthVendingActor
+import org.scalatest.concurrent.Eventually
+import org.scalatest.flatspec.AnyFlatSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class GithubAuthVendingActorSpec extends TestKitSuite with AnyFlatSpecLike with Matchers with Eventually {
+
+  final private val githubAuthEnabledServiceConfig =
+    ConfigFactory.parseString(s"""
+                                 |enabled = true
+                                 |auth.azure = true
+                                 |ecm.base-url = "https://mock-ecm-url.org"
+        """.stripMargin)
+
+  behavior of "GithubAuthVendingActor"
+
+  it should "return NoGithubAuthResponse if GithubAuthVending is disabled" in {
```

Comment:
These also look like a series of very similar tests. Could these also be table based?

Reply:
I will take a stab and try and make them table based tests.

Reply:
The tests have been updated to be table based.